pkg/driver/oxide/oxide.go

```diff
@@ -0,0 +1,436 @@
+	d.VpcName = opts.String(flagVpc)
+	d.SubnetName = opts.String(flagSubnet)
+
+	d.BaseDriver.SSHUser = opts.String(flagSSHUser)
+	d.SSHKeys = opts.StringSlice(flagSSHPubKeys)
+
+	// base driver flags
+	d.BaseDriver.SSHPort = 22
+	d.SetSwarmConfigFromFlags(opts)
+
+	return
+}
+
+// PreCreateCheck is called to enforce pre-creation steps
+func (d *Driver) PreCreateCheck() error {
+	userData, err := d.userDataBase64()
+	if err != nil {
+		return err
+	}
+	if len(userData)*3/4 > 32767 {
+		return fmt.Errorf("user data is too large: 32k limit")
+	}
+	return nil
+}
+
+func (d *Driver) Create() (err error) {
+	var ctx = context.Background()
+	var networkInterfaces []oxide.InstanceNetworkInterface
+	var sshKey *oxide.SshKey
+	var userData string
+
+	if err = d.SetupClient(); err != nil {
+		return
+	}
+
+	if sshKey, err = d.SshSetup(); err != nil {
+		return
+	}
+
+	sshKeys := []oxide.NameOrId{
+		oxide.NameOrId(sshKey.Id),
+	}
+
+	for _, key := range d.SSHKeys {
+		sshKeys = append(sshKeys, oxide.NameOrId(key))
+	}
+
+	if userData, err = d.userDataBase64(); err != nil {
+		return
+	}
+
+	if _, err = d.Client.InstanceCreate(ctx,
+		oxide.InstanceCreateParams{
+			Project: oxide.NameOrId(d.Project),
+			Body: &oxide.InstanceCreate{
+				Description: "built by rancher-machine driver",
+				BootDisk: &oxide.InstanceDiskAttachment{
+					Description: "docker-machine disk",
+					DiskSource: oxide.DiskSource{
+						ImageId: d.DiskImageId,
+						Type:    oxide.DiskSourceTypeImage,
+					},
+					Name: oxide.Name(d.diskName()),
+					Size: oxide.ByteCount(d.DiskSize),
+					Type: oxide.InstanceDiskAttachmentTypeCreate,
+				},
+				Disks:    []oxide.InstanceDiskAttachment{},
+				Hostname: oxide.Hostname(d.MachineName),
+				Memory:   oxide.ByteCount(d.Memory),
+				Name:     oxide.Name(d.MachineName),
+				Ncpus:    oxide.InstanceCpuCount(d.Ncpus),
+				NetworkInterfaces: oxide.InstanceNetworkInterfaceAttachment{
+					Type: oxide.InstanceNetworkInterfaceAttachmentTypeCreate,
+					Params: []oxide.InstanceNetworkInterfaceCreate{
+						{
+							Description: "created by the rancher-machine driver",
+							Name:        oxide.Name("rancher-driver-" + d.MachineName),
+							SubnetName:  oxide.Name(d.SubnetName),
+							VpcName:     oxide.Name(d.VpcName),
+						},
+					},
+				},
+				UserData:      userData,
+				SshPublicKeys: sshKeys,
+			},
+		},
+	); err != nil {
+		return
+	}
+
+	if networkInterfaces, err = d.Client.InstanceNetworkInterfaceListAllPages(ctx,
+		oxide.InstanceNetworkInterfaceListParams{
+			Project:  oxide.NameOrId(d.Project),
+			Instance: oxide.NameOrId(d.MachineName),
+		},
+	); err != nil {
+		return
+	}
+
+	for _, nic := range networkInterfaces {
+		// TODO: This is kind of an open question.
+		// The network interfaces come back in order?
+		// The IPAdress needs to be the one that the rancher server can hit.
+		// so if there are multiple NICs, we may be able to check the VPC
+		// it's attached or subnet or something like that..
```

Comment:
The NICs are listed in ascending alphabetical order. For a NIC to be added to an instance it must be from the same VPC as the other NICs, but all NICs must be from distinct subnets. There's also a concept of a "primary" NIC. By default this is the first NIC that was added, but it can be changed to another one

Reply:
Ah... ok. Maybe let's leave this comment here for now so we can readdress at a later date? I believe most of the time we'll just have a single NIC especially in the short term. We can add more functionality later. 
Thoughts?